api/src/main/java/com/google/appengine/api/mail/MailServiceImpl.java

```diff
@@ -48,13 +79,182 @@ public void send(Message message)
     doSend(message, false);
   }
 
+  private void sendSmtp(Message message, boolean toAdmin)
+      throws IllegalArgumentException, IOException {
+    String smtpHost = envProvider.getenv("SMTP_HOST");
+    if (smtpHost == null || smtpHost.isEmpty()) {
+      throw new IllegalArgumentException("SMTP_HOST environment variable is not set.");
+    }
+    Properties props = new Properties();
+    props.put("mail.smtp.host", smtpHost);
+    props.put("mail.smtp.port", envProvider.getenv("SMTP_PORT"));
+    props.put("mail.smtp.auth", "true");
+    if (Boolean.parseBoolean(envProvider.getenv("SMTP_USE_TLS"))) {
+      props.put("mail.smtp.starttls.enable", "true");
+    }
+
+    Session session = Session.getInstance(props, new javax.mail.Authenticator() {
+      protected javax.mail.PasswordAuthentication getPasswordAuthentication() {
+        return new javax.mail.PasswordAuthentication(
+            envProvider.getenv("SMTP_USER"), envProvider.getenv("SMTP_PASSWORD"));
```

Comment:
PASSWORD: does it mean it is in clear in config file? also visible when we dump env vars? 
Might need to find a more secure solution, maybe https://stackoverflow.com/questions/58371905/how-to-handle-secrets-in-google-app-engine or other best practice.